snapcraft/internal/lifecycle/_packer.py

```diff
@@ -37,15 +41,28 @@ def _snap_data_from_dir(directory):
         "version": snap["version"],
         "arch": snap.get("architectures", []),
         "type": snap.get("type", ""),
+        "license": snap.get("license"),
     }
 
 
 def pack(directory, output=None):
     mksquashfs_path = file_utils.get_tool_path("mksquashfs")
 
     snap = _snap_data_from_dir(directory)
-    output_snap_name = output or common.format_snap_name(snap)
+    # Verify that the snap command is available and use it to
+    # validate the layout.
+    # The snap command will most likely be found as it lives in
+    # core and the snapcraft snap lives on top of it (on the side
+    # rather).
+    if os.path.exists(_SNAP_PATH):
+        _run_snap_pack_verification(directory=directory),
+    if snap.get("license"):
+        logger.warning(
+            "Could not find {!r}, validity of the license string will only "
+            "take place once pushed to the store.".format(_SNAP_PATH)
```

Comment:
validity will be checked, or validation will take place, not validity will take place